File: snippets/fsharp/VS_Snippets_CLR_System/system.DateTime.Parse/fs/Parse1.fs
module Parse1

// <Snippet1>
open System
open System.Globalization

[<EntryPoint>]
let main _ =
    // Assume the current culture is en-US.
    // The date is February 16, 2008, 12 hours, 15 minutes and 12 seconds.

    // Use standard en-US date and time value
    let dateString = "2/16/2008 12:15:12 PM"
    try
        let dateValue = DateTime.Parse dateString
        printfn $"'{dateString}' converted to {dateValue}."
    with :? FormatException ->
        printfn $"Unable to convert '{dateString}'."

    // Reverse month and day to conform to the fr-FR culture.
    // The date is February 16, 2008, 12 hours, 15 minutes and 12 seconds.
    let dateString = "16/02/2008 12:15:12"
    try
        let dateValue = DateTime.Parse dateString
        printfn $"'{dateString}' converted to {dateValue}."
    with :? FormatException ->
        Console.WriteLine("Unable to convert '{0}'.", dateString)

    // Call another overload of Parse to successfully convert string
    // formatted according to conventions of fr-FR culture.
    try
        let dateValue = DateTime.Parse(dateString, CultureInfo("fr-FR", false))
        printfn $"'{dateString}' converted to {dateValue}."
    with :? FormatException ->
        printfn $"Unable to convert '{dateString}'."

    // Parse string with date but no time component.
    let dateString = "2/16/2008"
    try
        let dateValue = DateTime.Parse dateString
        printfn $"'{dateString}' converted to {dateValue}."
    with :? FormatException ->
        printfn $"Unable to convert '{dateString}'."

    0

// The example displays the following output to the console:
//       '2/16/2008 12:15:12 PM' converted to 2/16/2008 12:15:12 PM.
//       Unable to convert '16/02/2008 12:15:12'.
//       '16/02/2008 12:15:12' converted to 2/16/2008 12:15:12 PM.
//       '2/16/2008' converted to 2/16/2008 12:00:00 AM.
// </Snippet1>

File: snippets/csharp/System/String/Overview/parse1.cs
// <Snippet9>
using System;
using System.Globalization;

public class Example
{
    public static void Main()
    {
        string dateString = "07/10/2011";
        CultureInfo[] cultures = [CultureInfo.InvariantCulture,
                                  CultureInfo.CreateSpecificCulture("en-GB"),
                                  CultureInfo.CreateSpecificCulture("en-US")];
        Console.WriteLine($"{"Date String",-12} {"Culture",10} {"Month",8} {"Day",8}\n");
        foreach (var culture in cultures)
        {
            DateTime date = DateTime.Parse(dateString, culture);
            Console.WriteLine($"{dateString,-12} {(string.IsNullOrEmpty(culture.Name) ?
                              "Invariant" : culture.Name),10} {date.Month,8} {date.Day,8}");
        }
    }
}
// The example displays the following output:
//       Date String     Culture    Month      Day
//
//       07/10/2011    Invariant        7       10
//       07/10/2011        en-GB       10        7
//       07/10/2011        en-US        7       10
// </Snippet9>
